quiz/backend/stats/models.py

```diff
@@ -0,0 +1,75 @@
+from django.db import models
+
+from helpers.validators.validators import validate_badge_file_type
+
+
+class TimestampModel(models.Model):
+    """Abstract class for saving timestamps.
+
+    :param created_at: Date and time when given object was created.
+    :type created_at: models.DateTimeField(auto_now_add=True)
+
+    :param updated_at: Date and time of when given object was updated.
+    :type updated_at: models.DateTimeField(auto_now_add=True)
+
+    """
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+
+
+class Badge(TimestampModel):
+    """Represents badges that players/users can earn during games.
+    Inherits from TimestampModel and :class:`stats.models.TimestampModel`
+
+    Following fields are inherited:
+        * created_at
+        * updated_at
+
+    :param name: Name of badge
+    :type name: models.CharField(max_length=40, unique=True)
+
+    :param description: Description of bade. For e.g. when user/player can obtain it.
+    :type description: models.CharField(max_length=300)
+
+    :param owned_by_user_percentage: How many users/players own given badge.
+    :type owned_by_user_percentage: models.FloatField(default=0)
+
+    :param weight: Base for calculating bonus points after player/user receives given badge.
+    :type weight: models.PositiveSmallIntegerField(default=1)
+
+    :param points: How many points is given badge worth.
+    :type points: models.PositiveSmallIntegerField(default=0)
+
+    :param created_at: Date and time when badge was created.
+    :type created_at: models.DateTimeField(auto_now_add=True)
+
+    :param updated_at: Date and time when badge was updated, changes every time badge is edited.
+    :type updated_at: models.DateTimeField(auto_now=True)
+
+    """
+
+    name = models.CharField(max_length=40, unique=True)
+    description = models.CharField(max_length=300)
+    owned_by_user_percentage = models.FloatField(
+        default=0, help_text="Percentage of all users/players owning given badge."
+    )
+    weight = models.PositiveSmallIntegerField(
```

Comment:
I'm still unclear about need for points / weight for a given badge.

Reply:
Well I thought that @SlawCzech explained it, and he designed the system that is only based on Elo. @SlawCzech back me up😅

Reply:
We can give up on weight, but the field for points is really necessary. Badges will prop up users' stats to encourage them to compete and gain even more points. Gaining a badge will add extra points to user's account (like on codewars for instance).

Reply:
So long story short we don't have a simple Elo system, but something based one Elo, and `weight` and `points` of badge are a part of it. As I understand.